packages/core-data/src/selectors.ts

```diff
@@ -891,6 +891,57 @@ export const getEditedEntityRecord = createSelector(
 	}
 );
 
+/**
+ * Returns a list of entity records, merged with their edits.
+ *
+ * @param state     State tree.
+ * @param kind      Entity kind.
+ * @param name      Entity name.
+ * @param recordIds Record IDs.
+ *
+ * @return The list of entity records, merged with their edits.
+ */
+export const getEditedEntityRecords = createSelector(
```

Comment:
Let's make this new selector private for now.

Reply:
Done: https://github.com/WordPress/gutenberg/pull/67872/commits/e4cd460ee05bb514677af29f12a1492721960843